### ISSUE TITLE
encoding then decoding fails with certain strings

### ISSUE THREAD

**Issue by armijnhemel** [TkTech/mutf8#1]
The following string (from ICU4J 4) cannot be encoded and then decoded:

```
>>> a = u"[\uAC00 \uB098 \uB2E4 \uB77C \uB9C8 \uBC14  \uC0AC  \uC544 \uC790  \uCC28 \uCE74 \uD0C0 \uD30C \uD558]"
>>> a
'[가 나 다 라 마 바  사  아 자  차 카 타 파 하]'
>>> b = mutf8.encode_modified_utf8(a)
>>> b
b'[\xea\xb0\x80 \xeb\x82\x98 \xeb\x8b\xa4 \xeb\x9d\xbc \xeb\xa7\x88 \xeb\xb0\x94  \xec\x82\xac  \xec\x95\x84 \xec\x9e\x90  \xec\xb0\xa8 \xec\xb9\xb4 \xed\x83\x80 \xed\x8c\x8c \xed\x95\x98]'
>>> mutf8.decode_modified_utf8(b)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeDecodeError: 'mutf-8' codec can't decode byte 0xed in position 56: 6-byte codepoint started, but input too short to finish.

```

**Comment by armijnhemel** [comment]
As a follow up: encoding works correctly. The result is the same that I see in a Dalvik file. It is decoding that doesn't seem to work correctly.

**Comment by TkTech** [comment]
Thanks for the report @armijnhemel, taking a look.

**Comment by TkTech** [comment]
Found two issues, one in the Python implementation and one in c. Working on the fix.

**Comment by armijnhemel** [comment]
Some other byte strings that cannot be decoded in case you need more things to test with:

```
b'Lo/\xed\x80\xad;'
b'Lo/\xed\x82\x91;'
b'Lo/\xed\x9d\x80;'
```

**Comment by TkTech** [comment]
Sorry for the delay, went on a whale watching tour :)

@armijnhemel - can you give the `issue_1` branch a try?

**Comment by armijnhemel** [comment]
> Sorry for the delay, went on a whale watching tour :)

Magnificent animals, so I will allow it ;-)

> @armijnhemel - can you give the `issue_1` branch a try?

Seems to work on my test cases so far.

**Comment by TkTech** [comment]
Currently passing on everything I can find, so it'll be released tomorrow.

**Comment by TkTech** [comment]
Released as 1.0.4, thanks for reporting.
